### ISSUE TITLE
Add "real" ggplot examples to `label_*()` and `breaks_*()` functions

### ISSUE THREAD

**Issue by mine-cetinkaya-rundel** [r-lib/scales#419]
Currently the examples in the docs for `label_*()` and `breaks_*()` functions use `demo_*()` to showcase what various functions do, but they don't exemplify how to use them in the `scale_*()` layer of a ggplot. It would be good to add a simple, real plotting example as the first example in each of these functions. Then, the demo functions can be left as is to avoid code duplication to showcase various features and for other (testing, etc.) benefits these functions might have.

**Comment by cynthiahqy** [comment]
I'm picking this up for tidy dev day. After chatting with @thomasp85, the plan is to add one very simple ggplot2 example to each `label_*()` / `breaks_*()` doc page, as the first example. The existing `demo_*()` examples stay as they are.

Several functions share a page via `@rdname`, so there are **22 doc pages** to update:

### Breaks

| Page | Source | Also covers |
|---|---|---|
| `breaks_width()` | `R/breaks.R` | |
| `breaks_extended()` | `R/breaks.R` | |
| `breaks_pretty()` | `R/breaks.R` | |
| `breaks_timespan()` | `R/breaks.R` | |
| `breaks_exp()` | `R/breaks.R` | |
| `breaks_log()` | `R/breaks-log.R` | |
| `minor_breaks_log()` | `R/breaks-log.R` | |
| `minor_breaks_width()` | `R/minor_breaks.R` | `minor_breaks_n()` |

### Labels

| Page | Source | Also covers |
|---|---|---|
| `label_number()` | `R/label-number.R` | `label_comma()` |
| `label_currency()` | `R/label-currency.R` | |
| `label_percent()` | `R/label-percent.R` | |
| `label_bytes()` | `R/label-bytes.R` | |
| `label_date()` | `R/label-date.R` | `label_date_short()`, `label_time()`, `label_timespan()` |
| `label_dictionary()` | `R/label-dictionary.R` | |
| `label_parse()` | `R/label-expression.R` | `label_math()` |
| `label_glue()` | `R/label-glue.R` | |
| `label_log()` | `R/label-log.R` | |
| `label_number_auto()` | `R/label-number-auto.R` | |
| `label_ordinal()` | `R/label-ordinal.R` | |
| `label_pvalue()` | `R/label-pvalue.R` | |
| `label_scientific()` | `R/label-scientific.R` | |
| `label_wrap()` | `R/label-wrap.R` | |

### Notes
- **Skipping** `label_number_si()` (deprecated, internal) and `label_dollar()` (superseded by `label_currency()`).
- `label_date()` already has a ggplot2 example further down the page. I'll move it to the top.
- `minor_breaks_width()` still has an `if (FALSE)` example waiting on tidyverse/ggplot2#3591. I'll replace it with a working example.
- New examples will use `@examplesIf requireNamespace("ggplot2", quietly = TRUE)`, since ggplot2 is only in Suggests.
- Breaks examples will set `breaks` only (default labels are enough to show the effect), with two exceptions:
  - `breaks_timespan()` will be paired with `label_timespan()`; otherwise the axis shows raw numbers and the unit-aware breaks aren't visible.
  - `minor_breaks_*()` examples need visible minor grid lines (default `theme_gray()`), and `minor_breaks_log()` needs a log scale with a wide range, e.g. `c(1, 1e6)`.

**Comment by cynthiahqy** [comment]
### Proposed datasets for the examples

*Recommendations drafted by Claude (Opus 5).* All examples will only use datasets from the base R [`datasets`](https://stat.ethz.ch/R-manual/R-devel/library/datasets/html/00Index.html) package.

#### Datasets to reuse

| Dataset | What it is | Good for |
|---|---|---|
| `mtcars` | Car specs, 32 cars | Simple numbers, counts, long car names |
| `faithful` | Old Faithful: minutes between eruptions (43–96) | Plain breaks, minutes as a timespan |
| `airquality` | New York, May–Sep 1973 | Dates (built from `Month` and `Day`) |
| `beaver1` | Beaver body temperature, time as `hhmm` | Times of day |
| `islands` | Land mass areas, 12 to 16,988 thousand sq mi | Log scales, scientific notation |
| `state.x77` | US state stats (Area up to 566,432; Income $3,098–6,315) | Commas, currency |
| `ToothGrowth` | Supplement codes `OJ` / `VC` | Replacing codes with names |
| `PlantGrowth` | Groups `ctrl`, `trt1`, `trt2` | Math and parsed labels |
| `iris` | Species names | Glue labels |

#### Proposed dataset for each page

| Page | Dataset | Idea |
|---|---|---|
| `breaks_width()` | `faithful` | `waiting` with breaks every 5 minutes |
| `breaks_extended()` | `mtcars` | `mpg` with `breaks_extended(n = 10)` |
| `breaks_pretty()` | `airquality` | Date axis with `breaks_pretty(n = 10)` |
| `breaks_timespan()` | `faithful` | `waiting` with `breaks_timespan("mins")` + `label_timespan(unit = "mins")` |
| `breaks_exp()` | `faithful` | `eruptions` (1.6–5.1) with `transform = "exp"` |
| `breaks_log()` | `islands` | Area on `scale_x_log10()` |
| `minor_breaks_log()` | `islands` | Same plot with `minor_breaks = minor_breaks_log()` |
| `minor_breaks_width()` | `faithful` | `waiting` with `minor_breaks_width(1)` |
| `label_number()` | `state.x77` | Population (thousands) with `scale_cut = cut_short_scale()` |
| `label_currency()` | `state.x77` | Income per capita |
| `label_percent()` | `mtcars` | Bars of `cyl` by `gear` with `position = "fill"` |
| `label_bytes()` | `object.size()` of datasets | In-memory size of `mtcars`, `iris`, `airquality`, `faithful`, `quakes` |
| `label_date()` | `airquality` | Ozone by date with `label_date("%b %d")`, plus `label_date_short()` |
| `label_dictionary()` | `ToothGrowth` | `c(OJ = "Orange juice", VC = "Vitamin C")` |
| `label_parse()` | `PlantGrowth` | Levels renamed to `trt[1]`, `trt[2]` |
| `label_glue()` | `iris` | `label_glue("I. {x}")` → *I. setosa* |
| `label_log()` | `islands` | `scale_x_log10(labels = label_log())` |
| `label_number_auto()` | `islands` | Area × 1000 in sq mi |
| `label_ordinal()` | `mtcars` | Bar chart of `gear` → 3rd, 4th, 5th |
| `label_pvalue()` | `mtcars` | P-values from `cor.test()` of `mpg` vs other columns |
| `label_scientific()` | `islands` | Area × 1000 in sq mi |
| `label_wrap()` | `mtcars` | `mpg` by car name, wrapped at 10 characters |

#### Setup notes
- `state.x77` is a matrix → `as.data.frame()`
- `islands` is a named vector → `data.frame(area = islands)`
- `airquality` dates → `as.Date(paste(1973, Month, Day, sep = "-"))`
- `beaver1$time` → `as.POSIXct(sprintf("%04d", time), format = "%H%M")`
- `label_bytes()` sizes vary slightly across R versions, which is fine for an example.

**Comment by cynthiahqy** [comment]
### Proposed plot type for each example

*Suggestions drafted by Claude (Opus 5).* This builds on the dataset table above.

#### Rules for the examples
1. **The scale being shown decides the plot.** Use the simplest geom that puts that variable on an axis with enough breaks or labels to see the effect.
2. **The variable type decides the geom:**
   - one continuous variable → `geom_histogram()`
   - two continuous variables → `geom_point()`
   - a discrete variable → `geom_col()` / `geom_bar()` / `geom_boxplot()`
   - values over time → `geom_line()`
3. **Use the x axis by default**, to match the `demo_*()` examples that follow. Use y only when the value naturally sits there.
4. **Keep each example to 3–4 lines.** No facets, no themes, no colour unless needed, no custom stat settings.
5. **Set only `breaks` in breaks examples.** The only exception is `breaks_timespan()`, which is paired with `label_timespan()`.
6. **Guard each example** with `@examplesIf requireNamespace("ggplot2", quietly = TRUE)`, placed first, before the existing `demo_*()` examples.

#### Plan

| Page | Plot | Scale shown |
|---|---|---|
| `breaks_width()` | `faithful`: histogram of `waiting` | x |
| `breaks_extended()` | `mtcars`: point, `wt` vs `mpg` | x (`mpg`) |
| `breaks_pretty()` | `airquality`: line, `Temp` over date | x (date) |
| `breaks_timespan()` | `faithful`: histogram of `waiting` | x |
| `breaks_exp()` | `faithful`: point, `eruptions` vs `waiting` | x (`eruptions`) |
| `breaks_log()` | `islands`: histogram of area | x (log) |
| `minor_breaks_log()` | `islands`: histogram of area | x (log) |
| `minor_breaks_width()` | `faithful`: point, `waiting` vs `eruptions` | x (`waiting`) |
| `label_number()` | `state.x77`: point, `Population` vs `Income` | x (`Population`) |
| `label_currency()` | `state.x77`: point, `Income` vs `Life Exp` | x (`Income`) |
| `label_percent()` | `mtcars`: bar of `cyl` filled by `gear`, `position = "fill"` | **y** |
| `label_bytes()` | dataset sizes: `geom_col()`, names on y | x (sizes) |
| `label_date()` | `airquality`: line, `Ozone` over date | x |
| `label_dictionary()` | `ToothGrowth`: boxplot of `len` by `supp` | x (discrete) |
| `label_parse()` | `PlantGrowth`: boxplot of `weight` by `group` | x (discrete) |
| `label_glue()` | `ChickWeight`: boxplot of `weight` by `Diet` | x (discrete) |
| `label_log()` | `islands`: histogram of area | x (log) |
| `label_number_auto()` | `islands`: histogram of area × 1000 | x |
| `label_ordinal()` | `mtcars`: `geom_bar()` of `gear` | x |
| `label_pvalue()` | `mtcars` p-values: `geom_col()`, names on y | x (p-values) |
| `label_scientific()` | `islands`: histogram of area × 1000 | x |
| `label_wrap()` | `mtcars`: `geom_col()` of `mpg` by car name | x (discrete) |

#### Notes
- **Log scales:** `islands` has only one variable, so a histogram is the natural choice. A dot plot of the 48 land masses would show every value but is noisier.
- **`label_bytes()` / `label_pvalue()`:** horizontal bars keep the names readable, and the formatted values stay on x.
- **`label_wrap()`:** car names stay on x, where long labels collide and wrapping is needed. Use `head(mtcars, 8)` to avoid 32 crowded bars.
- **`label_ordinal()`:** `gear` is continuous, so the example needs `scale_x_continuous(breaks = 3:5, labels = label_ordinal())` to avoid breaks like 3.5.
- **`label_dictionary()`:** the dictionary is written inline, since there are only two entries:
  ```r
  ggplot(ToothGrowth, aes(supp, len)) +
    geom_boxplot() +
    scale_x_discrete(labels = label_dictionary(c(OJ = "Orange juice", VC = "Vitamin C")))
  ```
  The existing `lut` on that page is defined later and uses `mpg` drive codes, which aren't in `datasets`.
